packages/synapse-react-client/src/components/dataaccess/SubmissionPage.tsx

```diff
@@ -166,11 +176,11 @@ export default function SubmissionPage(props: SubmissionPageProps) {
         }}
       />
       <RejectDataAccessRequestModal
-        // Use the submission ID as a key so that the modal form is reset for a new submission
-        key={submissionId}
+        // Use the rejectionDialogKey as a key so that the modal form is reset each time its opened
+        key={rejectionDialogKey}
         submissionId={submissionId}
         open={showRejectionDialog}
-        onClose={() => setShowRejectionDialog(false)}
+        onClose={() => onCloseRejectionDialog()}
       />
```

Comment:
I think it would be more straightforward to remove the key and just render the jsx using 
{showRejectionDialog && <RejectDataAccessRequestModal
  ...
/>}

but please keep the comment!